### ISSUE TITLE
Stop parsing invalid dates

### ISSUE THREAD

**Issue by Allactaga** [scrapinghub/dateparser#66]
We should not really parse dates like this

```python
>>> from dateparser import parse
>>> parse("2015-03-17T16:37:51+00:002015-03-17T15:24:37+00:002015-03-17T15:02:08+00:002015-03-17T13:09:31+00:002015-03-17T11:34:21+00:002015-03-16T17:49:15+00:002015-03-16T17:33:30+00:002015-03-16T16:49:46+00:002015-03-16T15:50:57+00:002015-03-16T13:26:50+00:00 ")
datetime.datetime(2015, 3, 17, 13, 26, 50)
```

**Comment by MojoJolo** [comment]
Hi @Allactaga, I'm planning to get involve in this project.

To start off, what should we do if we saw invalid dates? Should we throw an exception?

**Comment by Allactaga** [comment]
Hi @MojoJolo, historically we've been simply returning `None` when we can't parse a string. We are not refusing possibility to change this bahavior to raising Exception in the future, but we'd need to plan a transition of our users. However about this particular issue, we've discussed it internally with @waqasshabbir and @asadurski and it is not a trivial thing to do. Also Waqas is assigned here.

How deeply you are planning to get involved? We are currently preparing new fix version with https://github.com/scrapinghub/dateparser/pull/86. Also for the next version you can target https://github.com/scrapinghub/dateparser/issues/54 or python 3 compatibility.

**Comment by MojoJolo** [comment]
@Allactaga Thanks, I'm checking https://github.com/scrapinghub/dateparser/issues/54. I think I can probably help in it.

**Comment by waqasshabbir** [comment]
Fixed in #187
